Subject: uitests: Fix with GtkFileChooserNative
From: Cole Robinson crobinso@redhat.com Sun Apr 7 15:12:49 2024 -0400
Date: Sun Apr 7 15:55:48 2024 -0400:
Git: 738a8d9f54e895e7de42d63d17a08a2aaa0a1f49

Bring back the `Open` label for `Browse Local` option, but
preserver the SAVE dialog behavior (can enter a manual path)

Signed-off-by: Cole Robinson <crobinso@redhat.com>

diff --git a/virtManager/storagebrowse.py b/virtManager/storagebrowse.py
index c5a26519..adc7857f 100644
--- a/virtManager/storagebrowse.py
+++ b/virtManager/storagebrowse.py
@@ -22,12 +22,14 @@ class _BrowseReasonMetadata:
         self.local_title = None
         self.gsettings_key = None
         self.dialog_type = None
+        self.choose_label = None
 
         if browse_reason == vmmStorageBrowser.REASON_IMAGE:
             self.enable_create = True
             self.local_title = _("Locate existing storage")
             self.storage_title = _("Locate or create storage volume")
             self.dialog_type = Gtk.FileChooserAction.SAVE
+            self.choose_label = _("_Open")
             self.gsettings_key = "image"
 
         if browse_reason == vmmStorageBrowser.REASON_ISO_MEDIA:
@@ -185,7 +187,8 @@ class vmmStorageBrowser(vmmGObjectUI):
         filename = self.err.browse_local(
             dialog_type=data.dialog_type,
             dialog_name=data.local_title,
-            start_folder=start_folder)
+            start_folder=start_folder,
+            choose_label=data.choose_label)
 
         if not filename:
             return
